### PR TITLE
Improve 'developing' instructions

### PR DESCRIPTION
**Pull request by tetchel**
As more team members have started running the vs code extension in Extension Development mode, it has become clear that our developing documentation needs work. I've added some new detail, and improved the overall flow based on feedback from team members.

I also separated it into its own document, so that it doesn't pollute our marketplace README.

### CODE REVIEW

**Comment by tetchel** [comment]
to make review easier; everything under `You can build the extension `.vsix` yourself` is unchanged.

**Comment by sishida** [comment]
@tetchel Do you know if we ever linked specifically to the "Developing Codewind for VS Code" section anywhere? If so, we need to update the link to this new page instead.

**Comment by tetchel** [comment]
I don't think we'd have that link anywhere but here.

File: DEVELOPING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+## Developing Codewind for VS Code
+1. First, clone this repository and download our dependencies.
```
Comment (sishida) [inline DEVELOPING.md]: Change the "our" to "the" to omit the use of first person.

**Comment by sishida** [comment]
I am approving this PR so that it can be merged. Then, I will add a new PR for ID edits.

**Pull request merged**